compiler/luci-interpreter/src/kernels/DepthwiseConv2D.cpp

```diff
@@ -171,5 +182,91 @@ void DepthwiseConv2D::evalQuantized() const
       getTensorShape(output()), getTensorData<uint8_t>(output()));
 }
 
+void DepthwiseConv2D::evalQuantizedS16() const
+{
+  const auto *input_data = getTensorData<int16_t>(input());
+  const auto *filter_data = getTensorData<int16_t>(filter());
+  const auto *bias_data = getTensorData<int64_t>(bias());
+  auto *output_data = getTensorData<int16_t>(output());
+
+  const Shape &input_shape = input()->shape();
+  const Shape &filter_shape = filter()->shape();
+  const Shape &output_shape = output()->shape();
+
+  const int32_t batches = input_shape.dim(0);
+  const int32_t input_height = input_shape.dim(1);
+  const int32_t input_width = input_shape.dim(2);
+  const int32_t input_depth = input_shape.dim(3);
+  const int32_t filter_height = filter_shape.dim(1);
+  const int32_t filter_width = filter_shape.dim(2);
+  const int32_t output_height = output_shape.dim(1);
+  const int32_t output_width = output_shape.dim(2);
+
+  const int32_t stride_height = _params.stride_height;
+  const int32_t stride_width = _params.stride_width;
+  const int32_t dilation_height_factor = _params.dilation_height_factor;
+  const int32_t dilation_width_factor = _params.dilation_width_factor;
+  const int32_t depth_multiplier = _params.depth_multiplier;
+
+  const double effective_output_scale =
+      getQuantizedConvolutionMultipler(input()->scale(), filter()->scale(), output()->scale());
```

Comment:
This assumes per-layer quantization. Can you add a kernel for per-channel quantization?

If we have a per-channel kernel, per-layer kernel may be easily replaced with the per-channel kernel (by calling per-channel kernel with the same scales/zero_points for all channels).

Reply:
FYI, we can check whether dw_conv is quantized per-channel or not by checking if the number of scales/zero_points of a filter is greater than 1.

Reply:
> This assumes per-layer quantization. Can you add a kernel for per-channel quantization?

There are no kernels with per-channel quantization support yet (even U8).
Per-channel quantization support should be more or less straightforward, but it is still a big work. It also does not seem that urgent, so I'd like to proceed as it is for now, if you don't mind.

Reply:
> It also does not seem that urgent, so I'd like to proceed as it is for now, if you don't mind.

I don't mind :) This is not urgent as you said.